tests/Integration/OptionsTest.php

```diff
@@ -0,0 +1,169 @@
+	 * @uses \Parsely\Parsely::get_managed_credentials
+	 */
+	public function test_set_default_track_as_values_should_not_be_called_when_saved_options_exist(): void {
+		$options                     = self::$parsely->get_options();
+		$options['track_post_types'] = array( 'new_post_type' );
+		$options['track_page_types'] = array( 'new_page_type' );
+
+		add_option( Parsely::OPTIONS_KEY, $options );
+		$saved_options = self::$parsely->get_options();
+
+		self::assertSame( array( 'new_post_type' ), $saved_options['track_post_types'] );
+		self::assertSame( array( 'new_page_type' ), $saved_options['track_page_types'] );
+
+		delete_option( Parsely::OPTIONS_KEY );
+	}
+
+	/**
+	 * Verifies that get_options() returns the correct "Track As" defaults when
+	 * Custom Post Types are registered.
+	 *
+	 * @since 3.9.0
+	 *
+	 * @covers Parsely\Parsely::get_options
+	 * @uses \Parsely\Parsely::get_managed_credentials
+	 * @uses \Parsely\Parsely::set_default_track_as_values
+	 */
+	public function test_get_options_track_as_defaults_when_cpts_are_registered(): void {
+		$custom_post_types = array(
+			// Post.
+			'custom_post_type'     => array(
+				'public'       => true,
+				'supports'     => array( 'editor' ),
+				'hierarchical' => false,
+			),
+			// Private post.
+			'excluded_post_type_1' => array(
+				'public'       => false,
+				'supports'     => array( 'editor' ),
+				'hierarchical' => false,
+			),
+			// Post without editor support.
+			'excluded_post_type_2' => array(
+				'public'       => true,
+				'supports'     => array( 'editor' => false ),
+				'hierarchical' => false,
+			),
+			// Private post without editor support.
+			'excluded_post_type_3' => array(
+				'public'       => false,
+				'supports'     => array( 'editor' => false ),
+				'hierarchical' => false,
+			),
+			// Page.
+			'custom_page_type'     => array(
+				'public'       => true,
+				'supports'     => array( 'editor' ),
+				'hierarchical' => true,
+			),
+			// Private page.
+			'excluded_page_type_1' => array(
+				'public'       => false,
+				'supports'     => array( 'editor' ),
+				'hierarchical' => true,
+			),
+			// Page without editor support.
+			'excluded_page_type_2' => array(
+				'public'       => true,
+				'supports'     => array( 'editor' => false ),
+				'hierarchical' => true,
+			),
+			// Private page without editor support.
+			'excluded_page_type_3' => array(
+				'public'       => false,
+				'supports'     => array( 'editor' => false ),
+				'hierarchical' => true,
+			),
+		);
+
+		foreach ( $custom_post_types as $key => $value ) {
+			register_post_type( $key, $value );
+		}
+
+		$options = self::$parsely->get_options();
+
+		self::assertSame( array( 'post', 'custom_post_type' ), $options['track_post_types'] );
+		self::assertSame( array( 'page', 'custom_page_type' ), $options['track_page_types'] );
+
+		// Clean up.
+		foreach ( $custom_post_types as $key => $value ) {
+			unregister_post_type( $key );
+		}
```

Comment:
Alternative option that's cleaner: add a tear_down method that calls `$this->reset_post_types()`

https://core.trac.wordpress.org/browser/trunk/tests/phpunit/includes/abstract-testcase.php#L294

Reply:
Done.